system/util/dummy_diag_publisher/CMakeLists.txt

```diff
@@ -1,69 +1,33 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(dummy_diag_publisher)
 
-add_compile_options(-std=c++14)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  diagnostic_updater
-  dynamic_reconfigure
-  roscpp
-)
-
-find_package(fmt REQUIRED)
+### Dependencies
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-generate_dynamic_reconfigure_options(
-  cfg/DummyDiagPublisher.cfg
-)
+### Target executable
+set(DUMMY_DIAG_PUBLISHER_SRC
+  src/dummy_diag_publisher_node/dummy_diag_publisher_node.cpp)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  CATKIN_DEPENDS
-    diagnostic_updater
-    dynamic_reconfigure
-)
+set(DUMMY_DIAG_PUBLISHER_HEADERS
+  include/dummy_diag_publisher/dummy_diag_publisher_node.h)
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
-
-# Target
-## dummy_diag_publisher_node
-add_executable(dummy_diag_publisher
+ament_auto_add_executable(${PROJECT_NAME}
   src/dummy_diag_publisher_node/main.cpp
-  src/dummy_diag_publisher_node/dummy_diag_publisher_node.cpp
+  ${DUMMY_DIAG_PUBLISHER_SRC}
+  ${DUMMY_DIAG_PUBLISHER_HEADERS}
```

Comment:
I think we don't have to add headers when we use ament_auto_add_executables

Reply:
yes, it's probably my "fault". Adding headers was a recommendation to make life easier for IDEs but I see that at least clion is clever enough and doesn't need it. I will update my porting-doc recommendations. Then you can just write here

```cmake
ament_auto_add_executable(${PROJECT_NAME} src/dummy_diag_publisher_node/main.cpp src/dummy_diag_publisher_node/dummy_diag_publisher_node.cpp)
```

Reply:
Thanks, I'll update this. I noticed that in certain Autoware.Auto packages, the headers were also specified. It would be nice to maybe have these smaller style details ironed out so that all the packages have the same consistent style.